src/main/java/selenium/Driver.java

```diff
@@ -0,0 +1,26 @@
+package selenium;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import selenium.pages.LoginPageWithPF;
+
+public final class Driver {
+    private static Driver instance;
+    public WebDriver driver;
+
+    private Driver(WebDriver driver) {
+        this.driver = driver;
+    }
+
+    public static Driver getInstance() {
+        if (instance == null) {
+            instance = new Driver(new ChromeDriver());
+        }
+        return instance;
+    }
+
+    public static void close() {
+        instance.driver.quit();
+        instance = null;
```

Comment:
no need to make this reference as null after invoking this method each time

Reply:
> This reference is needed. As if there is no reference than the following error is received during a test run of more than one test: `org.openqa.selenium.NoSuchSessionException: Session ID is null. Using WebDriver after calling quit()?`

first of all this you have such issue due to wrong structure of your Singleton, fix your code according to my previous comments and than we will talk about this issue.

Reply:
Updated